src/question/tests.py

```diff
@@ -1,3 +1,255 @@
+        self.prepare()
+        self.client.force_login(self.voter)
+        self.client.put(self.url, {'user_vote': 'up'})
+        self.client.logout()
+        self.client.force_login(self.voter2)
+        self.client.put(self.url, {'user_vote': 'none'})
+        self.client.logout()
+
+        self.assertEqual(self.upvoters.all().count(), 1)
+        self.assertEqual(self.downvoters.all().count(), 0)
+        self.assertEqual(self.upvoters.filter(user=self.voter).count(), 1)
+        self.assertEqual(self.downvoters.filter(user=self.voter).count(), 0)
+        self.assertEqual(self.upvoters.filter(user=self.voter2).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter2).count(), 0)
+
+
+    def test_down_down(self):
+        self.prepare()
+        self.client.force_login(self.voter)
+        self.client.put(self.url, {'user_vote': 'down'})
+        self.client.logout()
+        self.client.force_login(self.voter2)
+        self.client.put(self.url, {'user_vote': 'down'})
+        self.client.logout()
+
+        self.assertEqual(self.upvoters.all().count(), 0)
+        self.assertEqual(self.downvoters.all().count(), 2)
+        self.assertEqual(self.upvoters.filter(user=self.voter).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter).count(), 1)
+        self.assertEqual(self.upvoters.filter(user=self.voter2).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter2).count(), 1)
+
+
+    def test_down_none(self):
+        self.prepare()
+        self.client.force_login(self.voter)
+        self.client.put(self.url, {'user_vote': 'down'})
+        self.client.logout()
+        self.client.force_login(self.voter2)
+        self.client.put(self.url, {'user_vote': 'none'})
+        self.client.logout()
+
+        self.assertEqual(self.upvoters.all().count(), 0)
+        self.assertEqual(self.downvoters.all().count(), 1)
+        self.assertEqual(self.upvoters.filter(user=self.voter).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter).count(), 1)
+        self.assertEqual(self.upvoters.filter(user=self.voter2).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter2).count(), 0)
+
+
+    def test_none_none(self):
+        self.prepare()
+        self.client.force_login(self.voter)
+        self.client.put(self.url, {'user_vote': 'none'})
+        self.client.logout()
+        self.client.force_login(self.voter2)
+        self.client.put(self.url, {'user_vote': 'none'})
+        self.client.logout()
+
+        self.assertEqual(self.upvoters.all().count(), 0)
+        self.assertEqual(self.downvoters.all().count(), 0)
+        self.assertEqual(self.upvoters.filter(user=self.voter).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter).count(), 0)
+        self.assertEqual(self.upvoters.filter(user=self.voter2).count(), 0)
+        self.assertEqual(self.downvoters.filter(user=self.voter2).count(), 0)
+
+
```

Comment:
Some more tests would be nice to test errors: a user trying to vote without being logged in, vote on a question that doesn't exist, etc.

Reply:
Added tests to test non-existent question and no login in https://github.com/DevTeamSCH/vikoverflow-backend/pull/12/commits/01b536a456a5d593cff71ce02dcf44d2a7b10a32